packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -42,25 +33,24 @@ const emptyParams = {};
  *     };
  */
 const usePermissions = <Permissions = any, Error = any>(
-    params = emptyParams
-): State<Permissions, Error> => {
-    const [state, setState] = useSafeSetState<State<Permissions, Error>>({
-        isLoading: true,
-    });
+    params = emptyParams,
+    useQueryParams: UseQueryOptions<Permissions, Error> = {
+        refetchOnMount: false,
+    }
+) => {
     const getPermissions = useGetPermissions();
```

Comment:
Since we don't need to make the effect dependent on `getPermissions` anymore, I'd use `useAuthProvider` instead of `useGetPermissions`.

Reply:
I removed the useGetPermissions and added a conditional if authProvider is null.
does that work for you?